src/plots/plot.ts

```diff
@@ -855,6 +865,41 @@ export class Plot extends Component {
     return nearest === undefined ? undefined : this._lightweightPlotEntityToPlotEntity(nearest);
   }
 
+  /**
+   * Returns the entites whose bounding boxes overlap the parameter.
+   */
+  public entitiesInBounds(queryBounds: IEntityBounds): Plots.IPlotEntity[] {
+    const found = this._getEntityStore().entitiesInBounds(queryBounds);
+    if (!found) {
+      return undefined;
+    }
+    return found.map((entity) => this._lightweightPlotEntityToPlotEntity(entity));
+  }
+
+  /**
+   * Returns the entites whose bounding boxes overlap the parameter on the
+   * x-axis.
+   */
+  public entitiesInXRange(queryBounds: IEntityBounds): Plots.IPlotEntity[] {
```

Comment:
I'd prefer not to have separate `range/xRange/yRange` endpoints. We can do `XRange` by passing in `Infinity` for the `y/height` on the generic Bounds object. Maybe it'd make more sense for `Bounds` to be two `Range` objects

Reply:
nah, that'd just add another semantically equiv interface.

I think different endpoints is really helpful. It allows the user of the API to know that it'll "do what it says on the tin" as opposed to having to think about how the same method will re-interpret their arguments at runtime.